backend/app/api/endpoints/external_tasks.py

```
@@ -0,0 +1,47 @@
+from app.models.task import TaskResource
+from app.models.user import User
+from app.schemas.shared_task import TaskShareResponse
+from app.services.shared_task import shared_task_service
+
+router = APIRouter()
+
+
+@router.post("/{task_id}/share", response_model=TaskShareResponse)
+def share_task_with_api_key(
+    task_id: int,
+    current_user: User = Depends(security.get_current_user_jwt_apikey_tasktoken),
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)